src/workerd/util/sqlite-kv.h

```diff
@@ -15,8 +17,22 @@ namespace {
 // In this case we just customize the error reporting to emit JSG user visible errors instead
 // of KJ exceptions which become internal errors.
 class SqliteKvRegulator: public SqliteDatabase::Regulator {
-  void onError(kj::StringPtr message) const override {
-    JSG_ASSERT(false, Error, message);
+  void onError(kj::Maybe<int> sqliteErrorCode, kj::StringPtr message) const override {
+    KJ_IF_SOME(ec, sqliteErrorCode) {
+      switch (ec) {
+        case SQLITE_TOOBIG:
+          // We want to return SQLITE_TOOBIG to the user since it's usually because of user error.
+          JSG_ASSERT(false, Error, message);
+          return;
+        // We don't want to return other errors to the user since they're usually our fault.
+        // In that case we do nothing because the contract of onError is not to handle the error in
+        // its entirity, but instead to optionally handle it, and do nothing otherwise.
+        // When onError does nothing, the code calling into onError is still responsible for
+        // handling the error by other means, usually by throwing a KJ exception itself.
+        default: {
```

Comment:
rather than doing this dance with the braces, just do

```
default:
    return;
```

You may also want to put a KJ_UNREACHABLE at the after the switch statement to be clear you're never supposed to get there.

Reply:
Changed it to your suggestion, but I don't mind the fall through behavior as much because other codepaths (when the error code is kj::none) already just fall through to the end of the method. It seems nice to share that fallthrough path even if it doesn't do anything right now.